hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+    exit 1
+}
+
+# echoes the command provided as $@ and then runs it
+echo_and_eval () {
+    echo "> $*"
+    echo ""
+    eval "$@"
+}
+
+# runs the command provided as $@, and either returns silently with
+# status 0 or else logs an error message with the command's output
+try_eval () {
+    tmpfile=`mktemp`
+    if ! eval "$@" >& $tmpfile; then
+	status=1
+	echo "ERROR: Could not run '$*':"
+	sed -e 's/^/  /' $tmpfile
+	echo ""
+    else
+	status=0
+    fi
+    rm -f $tmpfile
+    return $status
+}
+
+# The environment may contain sensitive information like passwords or private keys
+filter_env () {
+    awk '/ env:$/ { indent = index($0, "e"); skipping = 1; next } !skipping { print; } skipping { ch = substr($0, indent, 1); if (ch != " " && ch != "-") { skipping = 0; print } }'
+}
+
+log_service () {
+    logpath=$1
+    service=$2
+    start_args=$3
+
+    echo_and_eval  journalctl -u $service                         &> $logpath/journal-$service
+    echo_and_eval  systemctl show $service                        &> $logpath/systemctl-show-$service
+
+    config_file=$(get_config_path_from_service "$start_args" "$service")
+    if [ -f "$config_file" ]; then
+        echo_and_eval  cat $config_file                           &> $logpath/CONFIG-$service
+    fi
+}
+
+log_system () {
+    logpath=$1
+
+    echo_and_eval  journalctl --boot                                  &> $logpath/journal-full
+    echo_and_eval  nmcli --nocheck -f all dev show                    &> $logpath/nmcli-dev
+    echo_and_eval  nmcli --nocheck -f all con show                    &> $logpath/nmcli-con
+    echo_and_eval  head -1000 /etc/sysconfig/network-scripts/ifcfg-*  &> $logpath/ifcfg
+    echo_and_eval  ip addr show                                       &> $logpath/addresses
+    echo_and_eval  ip route show                                      &> $logpath/routes
+    echo_and_eval  ip neighbor show                                   &> $logpath/arp
+    echo_and_eval  iptables-save                                      &> $logpath/iptables
+    echo_and_eval  cat /etc/hosts                                     &> $logpath/hosts
+    echo_and_eval  cat /etc/resolv.conf                               &> $logpath/resolv.conf
+    echo_and_eval  lsmod                                              &> $logpath/modules
+    echo_and_eval  sysctl -a                                          &> $logpath/sysctl
+
+    echo_and_eval  oc version                                         &> $logpath/version
+    echo                                                             &>> $logpath/version
+    echo_and_eval  docker version                                    &>> $logpath/version
+    echo                                                             &>> $logpath/version
+    echo_and_eval  cat /etc/system-release-cpe                       &>> $logpath/version
+}
+
+do_master () {
+    if ! nodes=$(oc get nodes --template '{{range .items}}{{.spec.externalID}} {{end}}'); then
```

Comment:
Not obvious if this is checking:
 - content of `${nodes}`
 - return code of `${nodes}` assignment
 - return code of `oc get nodes`

Prefer to be more explicit